adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenNonInteractiveHandler.cs

```diff
@@ -126,8 +126,31 @@ public AcquireTokenNonInteractiveHandler(RequestData requestData, UserAssertion
                     {
                         throw new AdalException(AdalError.MissingFederationMetadataUrl);
                     }
-                    
-                    WsTrustAddress wsTrustAddress = await MexParser.FetchWsTrustAddressFromMexAsync(userRealmResponse.FederationMetadataUrl, this.userCredential.UserAuthType, RequestContext).ConfigureAwait(false);
+
+                    WsTrustAddress wsTrustAddress = null;
+                    try
+                    {
+                        var _wsTrustAddress = await Microsoft.Identity.Core.WsTrust.MexParser.FetchWsTrustAddressFromMexAsync(userRealmResponse.FederationMetadataUrl, this.userCredential.UserAuthType, RequestContext).ConfigureAwait(false);
+                        wsTrustAddress = new WsTrustAddress() { Uri = _wsTrustAddress.Uri, Version = _wsTrustAddress.Version == Identity.Core.WsTrust.WsTrustVersion.WsTrust13 ? WsTrustVersion.WsTrust13 : WsTrustVersion.WsTrust2005 };
```

Comment:
Style: avoid very long lines, makes the code hard to read and review because scrolling sideways is hard :).